<?xml version="1.0"?>

<launch>
      <arg name="robot_type" default="$(env ROBOT_TYPE)"
            doc="Robot type: [a1, aliengo, go1, laikago]" />

      <arg name="cheater" default="false" />

      <!-- make the files into global parameters -->
      <!-- <param name="urdfFile" value="/tmp/legged_control/$(arg robot_type).urdf" /> -->
      <!-- <param name="urdfFile" value="$(find legged_unitree_description)/urdf/go1_payload.urdf" -->
      <param name="urdfFile" value="$(find legged_unitree_description)/urdf/go1.urdf" />
      
      <!-- <param name="urdfFile" value="$(find legged_unitree_description)/urdf/go1.urdf" /> -->
      <param name="taskFile" value="$(find legged_controllers)/config/$(arg robot_type)/task.info" />
      <param name="referenceFile"
            value="$(find legged_controllers)/config/$(arg robot_type)/reference.info" />
      <param name="gaitCommandFile"
            value="$(find legged_controllers)/config/$(arg robot_type)/gait.info" />

      <rosparam file="$(find legged_controllers)/config/controllers.yaml" command="load" />

      <node if="$(arg cheater)" name="controller_loader" pkg="controller_manager"
            type="controller_manager"
            output="screen"
            args="load
          controllers/joint_state_controller
          controllers/legged_controller
          controllers/legged_cheater_controller
" />

      <node unless="$(arg cheater)" name="controller_loader" pkg="controller_manager"
            type="controller_manager"
            output="screen"
            args="load
          controllers/joint_state_controller
          controllers/legged_controller
" />

      <node pkg="ocs2_legged_robot_ros" type="legged_robot_gait_command"
            name="legged_robot_gait_command"
            output="screen" />

      <arg name="teleop_config" default="$(find legged_controllers)/config/rosjoy_key_map.yaml" />

      <rosparam file="$(arg teleop_config)" command="load" />
      <node pkg="joy" type="joy_node" name="joy_node" />

      <node pkg="legged_controllers" type="legged_target_trajectories_publisher"
            name="legged_robot_target"
            output="screen" />

      <node pkg="rosbag" type="record" name="record"
            args="record -o /home/ControlUser/legged_ctrl_ws/src/bags/ 
            /mocap_node/Go1_body/Odom
            /mocap_node/Box_body/Odom
            /unitree_hardware/imu
            /unitree_hardware/joint_foot
            /joint_controller_FL_calf/cmd 
            /joint_controller_FL_calf/state 
            /joint_controller_FL_hip/cmd 
            /joint_controller_FL_hip/state 
            /joint_controller_FL_thigh/cmd 
            /joint_controller_FL_thigh/state 
            /joint_controller_FR_calf/cmd 
            /joint_controller_FR_calf/state 
            /joint_controller_FR_hip/cmd 
            /joint_controller_FR_hip/state 
            /joint_controller_FR_thigh/cmd 
            /joint_controller_FR_thigh/state 
            /joint_controller_RL_calf/cmd 
            /joint_controller_RL_calf/state 
            /joint_controller_RL_hip/cmd 
            /joint_controller_RL_hip/state 
            /joint_controller_RL_thigh/cmd 
            /joint_controller_RL_thigh/state 
            /joint_controller_RR_calf/cmd 
            /joint_controller_RR_calf/state 
            /joint_controller_RR_hip/cmd 
            /joint_controller_RR_hip/state 
            /joint_controller_RR_thigh/cmd 
            /joint_controller_RR_thigh/state 
            /odom
        " />

</launch>